app/services/catalog/create_approval_request.rb

```diff
@@ -0,0 +1,62 @@
+module Catalog
+  class CreateApprovalRequest
+    attr_reader :order
+
+    def initialize(id)
+      @order = Order.find_by!(:id => id)
+    end
+
+    def process
+      @order.order_items.each do |order_item|
+        approvals = send_approval_requests(order_item)
+        approvals.compact.each do |req|
+          order_item.approval_requests << create_approval_request(req)
+        end
+      end
+
+      @order.update(:state => "Ordered", :ordered_at => Time.now.utc)
+      self
+    rescue Catalog::ApprovalError => e
+      Rails.logger.error("Error putting in approval Request for #{order.id}: #{e.message}")
```

Comment:
@lindgrenj6 This line is not being tested can you add a test for executing this code.

In your dev env if you run the following from a terminal you would see the lines not covered
```
export CI=1
bundle exec rake
open coverage/index.html
```

Reply:
Oh nice, I'll keep that env var set all the time.